jsco4180Sup/jsco4180.db

```diff
@@ -1006,18 +742,51 @@ record(bo, "$(P)ERROR:SP")
 
 alias("$(P)ERROR:SP", "$(P)ERROR:SP:RBV")
 
-record(stringin, "$(P)ERROR:STR")
+record(stringin, "$(P)ERROR:STR") 
 {
-    field(DESC, "Command input return error")
+    field(DESC, "Last error returned on input")
     field(DTYP, "stream")
     field(SCAN, "I/O Intr")
-    field(INP, "@jsco4180.proto readErrorString $(PORT)")
+    field(VAL, "No error")
+    
+    field(INP, "@jsco4180.proto readErrorStr $(PORT)")
 
+    info(archive, "VAL")
     field(SIML, "$(P)SIM")
     field(SIOL, "$(P)SIM:ERROR:STR")
     field(SDIS, "$(P)DISABLE")
 }
 
+record(bi, "$(P)ERROR:COMP") 
+{
+    field(DESC, "Device component error")
+    field(DTYP, "Soft Channel")
+    field(VAL, "0")
+    
+    field(ZNAM, "No issue")
+    
+    field(ONAM, "Single Channel")
+    field(OSV, "MAJOR")
+
+    info(archive, "VAL")
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:ERROR:COMP")
+    field(SDIS, "$(P)DISABLE")
+}
+
+record(ai, "$(P)ERROR:DELAY")
+{
+    field(DESC, "Delay before trying to reset components")
+    field(VAL, "1")
```

Comment:
This seems rather quick - the device is only scanning at 1s intervals. I think this should be quite significantly longer e.g. 15-30 seconds. This is reflected in the logs of the IOC test framework where it appears to be doing a reset every time the components change

Reply:
I think you may also need a `field(PINI, "YES")` for this to actually take effect